### PR TITLE
Add boids to examples

### PR DESCRIPTION
**Pull request by reczkok**
This is a slightly enhanced version of the boids from tutorial.

### CODE REVIEW

File: apps/typegpu-docs/src/content/examples/simulation/boids.ts [hunk 1]
```diff
@@ -0,0 +1,357 @@
...
+  }),
+  MAX_TRIANGLES,
+);
+type TrianglePosData = typeof trianglePosData;
+
+const trianglePosBuffers = Array.from({ length: 2 }, () => {
+  return wgsl.buffer(trianglePosData).$allowReadonly().$allowMutable();
+});
+
+const pairs = [
```
Comment (mhawryluk) [inline apps/typegpu-docs/src/content/examples/simulation/boids.ts]: i'm still not a fan of this variable name 😅

File: apps/typegpu-docs/src/content/examples/simulation/boids.ts [hunk 2]
```diff
@@ -0,0 +1,357 @@
...
+  cohesionDistance: 0.3,
+  cohesionStrength: 0.001,
+};
+
+function applyOptions() {
+  runtime.writeBuffer(parametesBuffer, options);
+}
+
+addSliderParameter(
+  'Separation Dist',
```
Comment (mhawryluk) [inline apps/typegpu-docs/src/content/examples/simulation/boids.ts]: all of our other examples have the parameter labels in lowercase, I would keep it consistent

File: apps/typegpu-docs/src/content/examples/simulation/boids.ts [hunk 3]
```diff
@@ -0,0 +1,357 @@
...
+
+const getRotationFromVelocity = wgsl.fn`(velocity: vec2f) -> f32 {
+  return -atan2(velocity.x, velocity.y);
+}`;
+
+const renderPipelines = [0, 1].map((idx) =>
+  runtime.makeRenderPipeline({
+    vertex: {
+      code: wgsl`
+      let triangleSize = ${triangleSizeBuffer.asUniform()};
```
Comment (mhawryluk) [inline apps/typegpu-docs/src/content/examples/simulation/boids.ts]: +1 to indentation of this wgsl code, please. same in fragment and compute

File: apps/typegpu-docs/src/content/examples/simulation/boids.ts [hunk 4]
```diff
@@ -0,0 +1,357 @@
...
+      cohesionStrength: f32,
+    }),
+    {
+      separationDistance: 0.03,
+      separationStrength: 0.001,
+      alignmentDistance: 0.3,
+      alignmentStrength: 0.01,
+      cohesionDistance: 0.3,
+      cohesionStrength: 0.001,
+    },
```
Comment (iwoplaza) [inline apps/typegpu-docs/src/content/examples/simulation/boids.ts:50]: Nitpick, seems like a duplicate of the `options` variable. Also, could we either rename `options` to `parameters` or `parametersBuffer` to `optionsBuffer` for consistency's sake?

**Pull request merged**